lib/switchtec.c

```diff
@@ -460,7 +479,14 @@ const char *switchtec_strerror(void)
 void switchtec_perror(const char *str)
 {
 	const char *msg = switchtec_strerror();
-	fprintf(stderr, "%s: %s\n", str, msg);
+
+	if (!strcmp(msg, "Unknown error"))
```

Comment:
This seems like you should have an if statement on the mrpc global variable. Why would "unknown error" indicate that you'd print with an MRPC CMD prefix?

Reply:
I wanted to give more detailed info which includes the CMD ID and error code returned from firmware for the *unknown MRPC errors*. So it seems I should return a more specific string from switchtec_strerror() to make it more clear, like 'Unknown MPRC error'.

For other cases, either proper platform error strings or strings for known MRPC errors are returned in switchtec_strerrro().

The reason to add this for unknown MRPC errors is that we found that it's likely some errors from firmware *may* not be documented in spec, so we cannot decode them properly. And for library user, it would be useful for them to show raw MRPC errors if they do not want the strings.